include/unifex/let.hpp

```diff
@@ -124,45 +124,38 @@ struct _predecessor_receiver<Operation>::type {
 
   template <typename... Values>
   void set_value(Values&&... values) && noexcept {
-    bool destroyedPredOp = false;
     auto& op = op_;
     try {
+      scope_guard destroyPredOp = [&]() noexcept { unifex::deactivate(op.predOp_); };
       auto& valueTuple =
-          op.values_.template get<decayed_tuple<Values...>>();
-      valueTuple.construct((Values &&) values...);
-      destroyedPredOp = true;
-      op.predOp_.destruct();
-      try {
-        auto& succOp =
-            op.succOp_.template get<successor_operation<Values...>>()
-                .construct_from([&] {
-                  return unifex::connect(
-                      std::apply(std::move(op.func_), valueTuple.get()),
-                      successor_receiver<Operation, Values...>{op});
-                });
-        unifex::start(succOp);
-      } catch (...) {
-        valueTuple.destruct();
-        throw;
-      }
+        op.values_.template construct<decayed_tuple<Values...>>((Values &&) values...);
+      destroyPredOp.reset();
+      scope_guard destroyValues = [&]() noexcept {
+        op.values_.template destruct<decayed_tuple<Values...>>();
+      };
+      auto& succOp =
+          unifex::activate_from<successor_operation<Values...>>(op.succOp_, [&] {
+              return unifex::connect(
+                  std::apply(std::move(op.func_), valueTuple),
+                  successor_receiver<Operation, Values...>{op});
+            });
+      unifex::start(succOp);
+      destroyValues.release();
```

Comment:
@kirkshoop, @lewissbaker: please take a hard look at this to make sure I preserved the logic.

Reply:
LGTM